src/main/java/com/kryptnostic/conductor/codecs/TimestampDateTimeTypeCodec.java

```diff
@@ -0,0 +1,68 @@
+package com.kryptnostic.conductor.codecs;
+
+import java.nio.ByteBuffer;
+
+import org.joda.time.DateTime;
+
+import com.datastax.driver.core.DataType;
+import com.datastax.driver.core.ParseUtils;
+import com.datastax.driver.core.ProtocolVersion;
+import com.datastax.driver.core.TypeCodec;
+import com.datastax.driver.core.exceptions.InvalidTypeException;
+
+/**
+ * TypeCodec between Cql Timestamp and Joda DateTime, modified from default TimestampCodec
+ * @author Ho Chung Siu
+ *
+ */
+public class TimestampDateTimeTypeCodec extends TypeCodec<DateTime> {
+    private static final TimestampDateTimeTypeCodec instance = new TimestampDateTimeTypeCodec();
+    
+    private TimestampDateTimeTypeCodec() {
+        super(DataType.timestamp(), DateTime.class);
+    }
+    
+    public static TimestampDateTimeTypeCodec getInstance(){
+        return instance;
+    }
+    
+    @Override
+    public DateTime parse(String value) {
+        if (value == null || value.isEmpty() || value.equalsIgnoreCase("NULL"))
+            return null;
+        // strip enclosing single quotes, if any
+        if (ParseUtils.isQuoted(value))
+            value = ParseUtils.unquote(value);
+
+        if (ParseUtils.isLongLiteral(value)) {
+            try {
+                return new DateTime(Long.parseLong(value));
+            } catch (NumberFormatException e) {
+                throw new InvalidTypeException(String.format("Cannot parse timestamp value from \"%s\"", value));
+            }
+        }
+
+        try {
+            return DateTime.parse( value );
+        } catch (IllegalArgumentException e) {
+            throw new InvalidTypeException(String.format("Cannot parse timestamp value from \"%s\"", value));
+        }
+    }
+
+    @Override
+    public String format(DateTime value) {
+        if (value == null)
+            return "NULL";
+        return Long.toString(value.getMillis());
```

Comment:
I'm not really sure when format gets called here as it is supposed to prefer ByteBuffer serialization, but based off the docs it looks like the CQL string literal it has to be in ISO 8601 format. It does say you can use an integer for input, but I think that's means actual integer not string. What do other codecs do?